File: fixtures/scale/ruby/large/lib/check_turnstile.rb
require_relative 'turnstile_encrypt'
require_relative 'turnstile_decrypt'
require_relative 'corpus_rotate'

def check_turnstile
  corpus_rotate.each do |text|
    sealed = turnstile_encrypt(text, 3)
    opened = turnstile_decrypt(sealed, 3)
    return false if opened != text
  end
  true
end

File: fixtures/scale/python/large/cipherlab/checks/check_turnstile.py
from cipherlab.ciphers.turnstile import turnstile_encrypt, turnstile_decrypt
from cipherlab.specs.turnstile_spec import turnstile_spec
from cipherlab.corpus.corpus_rotate import corpus_rotate


def check_turnstile():
    spec = turnstile_spec()
    for text in corpus_rotate():
        sealed = turnstile_encrypt(text, spec["key"])
        opened = turnstile_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "rotate"
